src/ProxyServer/ProxyServer.swift

```diff
@@ -29,6 +29,8 @@ open class ProxyServer: NSObject, TunnelDelegate {
     open var observer: Observer<ProxyServerEvent>?
 
     var tunnels: TunnelArray = []
+    
+    private let accessQueue = DispatchQueue(label: "NEKit.ProxyServer.TunnelArrayAccess", attributes: .concurrent)
```

Comment:
There is no need to use Atomic since everything is already protected. You can check that everything is running on the same serial dispatch queue.